spec/API_specification/creation_functions.md

```diff
@@ -33,11 +33,11 @@ This function cannot guarantee that the interval does not include the `stop` val
 
 -   **step**: _Union\[ int, float ]_
 
-    -   the distance between two adjacent elements (`out[i+1] - out[i]`). Default: `1`.
+    -   the distance between two adjacent elements (`out[i+1] - out[i]`). Must not be `0`; may be negative, this results in an empty array if `stop <= start`. Default: `1`.
 
 -   **dtype**: _Optional\[ &lt;dtype&gt; ]_
 
-    -   output array data type. If `dtype` is `None`, the output array data type must be the default floating-point data type. Default: `None`.
+    -   output array data type. If `dtype` is `None`, the output array data type must be inferred from `start`, `stop` and `step`. If those are all integers, the output array dtype must be the default integer dtype; if one or more have type `float`, then the output array dtype must be the default floating-point data type. Default: `None`.
```

Comment:
Does the spec talk about "default integer dtype" anywhere? It says

> A conforming implementation of the array API standard must define a default floating-point data type (either `float32` or `float64`), as well as a default data type for an array index (either `int32` or `int64`).

Reply:
No, not at the moment.

Reply:
It does, in the "data types" section. My last commit extends that paragraph.

Reply:
Meh, wording - that was intended as "default integer dtype" I believe.

Reply:
Updated that section. A default dtype for indexing arrays doesn't make sense to me; only the values indexing accepts are meaningful (can it exceed 32-bit). But given that we don't have advanced indexing, even that doesn't really make sense. Why do we need this at all?

Reply:
A default index data type stems from [`unique`](https://github.com/data-apis/array-api/blob/0cbbfead9ab7b4828921fbc01f40981d2702db39/spec/API_specification/set_functions.md#returns), where a list of indices is returned.

Reply:
We can modify to a more general "default integer data type", so long as an array of indices should have the same dtype as whatever is returned from `arange` when only integers are provided.

Reply:
`unique` should just use "default integer data type" indeed. Those returned indices are going to be fairly useless though if there's no integer array based indexing in https://data-apis.org/array-api/latest/API_specification/indexing.html.

Reply:
Not entirely useless. One can still manually iterate or perhaps provide to a separate function which performs the equivalent of integer array based indexing.

Reply:
This is what `take` can be used for, but we don't have it. On the other hand, if we'd be able to add `take`, why not just add 1-D integer array indexing?

Reply:
It would be easier to use `take` to be able to mark it as problematic for shape inference. Worth considering.

Reply:
Agreed.

Reply:
It's also relevant for integer indexing. A 32-bit integer index type limits the max size of a dimension.

Reply:
That's more a "how large can your array be" limitation than an indexing thing though.